### PR TITLE
[LTC] Upstream utils to extract BackendDevice from at::Tensor

### PR DESCRIPTION
**Pull request by alanwaketan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #70069
* #70066

Summary:
This commit upstreams utils to extract BackendDevice from at::Tensor.

Test Plan:
./build/bin/test_lazy --gtest_filter=BackendDeviceTest.GetBackendDevice*

Differential Revision: [D33293160](https://our.internmc.facebook.com/intern/diff/D33293160)

### CODE REVIEW

**Comment by alanwaketan** [comment]
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33172665).

File: torch/csrc/lazy/backend/backend_device.cpp [hunk 1]
```diff
@@ -53,5 +54,16 @@ c10::Device backendDeviceToAtenDevice(const BackendDevice& device) {
...
 
+c10::optional<BackendDevice> GetBackendDevice(const at::Tensor& tensor) {
+  if (auto lt = TryGetLtcTensor(tensor)) {
+    return lt.GetDevice();
+  }
+  return c10::nullopt;
+}
+
+c10::optional<BackendDevice> GetBackendDevice() {
+  return c10::nullopt;
```
Comment (wconstab) [inline torch/csrc/lazy/backend/backend_device.cpp:65]: What is the point of this function?

Comment (wconstab) [reply torch/csrc/lazy/backend/backend_device.cpp:65]: nevermind i see the comment in the header.

**Review by wconstab — APPROVED** [review]

File: torch/csrc/lazy/backend/backend_device.cpp (hunk 1 above)
Comment (alanwaketan) [reply torch/csrc/lazy/backend/backend_device.cpp:65]: Great!

**Comment by alanwaketan** [comment]
Thanks, Will.

**Comment by alanwaketan** [comment]
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33172665).

**Comment by alanwaketan** [comment]
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33172665).

**Comment by alanwaketan** [comment]
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33172665).

**Comment by suo** [comment]
Looks like this broke windows builds. http://hud2.pytorch.org/commit/121d06799954b0c0ba24f084f2d78b1a81a4e4ec I am reverting

**Comment by alanwaketan** [comment]
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33293160).

**Pull request closed without merging**